hermes/src/network/pythnet.rs

```diff
@@ -317,6 +330,115 @@ pub async fn spawn(opts: RunOptions, state: Arc<State>) -> Result<()> {
         })
     };
 
-    let _ = tokio::join!(task_listener, task_guadian_watcher);
+
+    let task_price_feeds_metadata_updater = {
+        let price_feeds_state = state.clone();
+        let price_feeds_update_interval = opts.price_feeds_cache_update_interval;
+        tokio::spawn(async move {
+            while !crate::SHOULD_EXIT.load(Ordering::Acquire) {
+                if let Err(e) =
+                    fetch_and_store_price_feeds_metadata(price_feeds_state.as_ref(), &rpc_client)
+                        .await
+                {
+                    tracing::error!("Error in fetching and storing price feeds metadata: {}", e);
+                }
+                // This loop with a sleep interval of 1 second allows the task to check for an exit signal at a
+                // fine-grained interval. Instead of sleeping directly for the entire `price_feeds_update_interval`,
+                // which could delay the response to an exit signal, this approach ensures the task can exit promptly
+                // if `crate::SHOULD_EXIT` is set, enhancing the responsiveness of the service to shutdown requests.
+                for _ in 0..price_feeds_update_interval {
+                    if crate::SHOULD_EXIT.load(Ordering::Acquire) {
+                        break;
+                    }
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                }
+            }
+        })
+    };
+
+    let _ = tokio::join!(
+        task_listener,
+        task_guardian_watcher,
+        task_price_feeds_metadata_updater
+    );
     Ok(())
 }
+
+
+pub async fn fetch_and_store_price_feeds_metadata(
+    state: &State,
+    rpc_client: &RpcClient,
+) -> Result<Vec<PriceFeedMetadata>> {
+    let price_feeds_metadata = fetch_price_feeds_metadata(&state, &rpc_client).await?;
+    store_price_feeds_metadata(&state, &price_feeds_metadata).await?;
+    Ok(price_feeds_metadata)
+}
+
+async fn fetch_price_feeds_metadata(
+    state: &State,
+    rpc_client: &RpcClient,
+) -> Result<Vec<PriceFeedMetadata>> {
+    let mut price_feeds_metadata = Vec::<PriceFeedMetadata>::new();
+    let mapping_address = &state.mapping_address;
+    let mapping_data = rpc_client.get_account_data(mapping_address).await?;
+    let mapping_acct = load_mapping_account(&mapping_data)?;
+
+    // Split product keys into chunks of 150 to avoid too many open files error (error trying to connect: tcp open error: Too many open files (os error 24))
+    for product_keys_chunk in mapping_acct
+        .products
+        .iter()
+        .filter(|&prod_pkey| *prod_pkey != Pubkey::default())
+        .collect::<Vec<_>>()
+        .chunks(150)
```

Comment:
tried with 200 but still intermittently getting `os error 24` so went with 150 to be safe